gnc/sub8_perception/sub8_vision_tools/machine_learning/verify_method_ada.py

```diff
@@ -0,0 +1,64 @@
+import cv2
+import numpy as np
+import pickle
+import argparse
+import sys
+from time import time
+from adaboost import observe
+from sklearn.ensemble import GradientBoostingClassifier
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(dest='pkl',
+                        help="The pkl file to test on")
+    parser.add_argument(dest='algorithm',
+                        help="The classifier to use")
+
+    args = parser.parse_args(sys.argv[1:])
+    data = pickle.load(open(args.pkl, "rb"))
+    clf = pickle.load(open(args.algorithm, 'rb'))
+
+    attributes = {
+        'times': [],
+        'true_positives': [],
+        'false_positives': [],
+    }
+    for u_image, u_mask in data:
+        image = u_image[::2, ::2, :]
+        mask = u_mask[::2, ::2]
+        tic_observation = time()
+        some_observations = observe(image)
+        print '-------------------------'
+        print 'Observing took {} seconds'.format(time() - tic_observation)
+        tic_prediction = time()
+        segmentation = [x for x in [clf.predict(obs.reshape(1,-1)) for obs in some_observations]]
```

Comment:
Adaboost has numpy support, so if you reshape the some_observations to (-1, num_features) you can predict them all at once

Will be much faster